src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/TargetingRulesTab.tsx

```diff
@@ -179,7 +208,24 @@ const TargetingRulesTab = ({
       initialValues={initialValues}
       validate={values => validate(values)}
       onSubmit={values => {
-        saveChanges(values)
+        if (initialValues.state === 'on') {
+          if (initialValues.onVariation != values.onVariation) {
+            setNewValues(values)
+            setVariationValue(values.onVariation)
+            openDialog()
+          } else {
+            saveChanges(values)
+          }
+        }
+        if (initialValues.state === 'off') {
```

Comment:
Can this just be `else`'d? It'll always be `on` or `off`, right?
```suggestion
        } else {
```

Reply:
Absolutely, have fixed